src/posthog-core.ts

```diff
@@ -232,6 +232,8 @@ export class PostHog {
     SentryIntegration: typeof SentryIntegration
     segmentIntegration: () => any
 
+    debugEventEmitter = new SimpleEventEmitter()
```

Comment:
Could we make this private? Feels like people might start using it otherwise. Ideally we move all `logger.info` stuff into this emitter instead (bigger change I realise) so that everything hooks into it including `.debug()` but I wouldn't want it to be exposed publicly. Better case would be like the js-lite sdk that we have a proper method like `posthog.on("error")` or something once we actually have everything under it.

Reply:
made `on` advertising only one event by contract

(will follow-up with moving other things onto it)